File: leetcode_learning/code_cpp/58.cpp
#include<iostream>
#include<string>

using namespace std;

class Solution {
public:
    int lengthOfLastWord(string s) {
        int index = s.size() - 1;
        while(s[index]==' ')
            index--;
        int result = 0;
        while(index>=0&&s[index]!=' ')
        {
            index--;
            result++;
        }
        return result;
    }
};

File: leetcode_learning/auto-leetcode/58.最后一个单词的长度/58.py
"""
58.最后一个单词的长度
难度：简单
链接：https://leetcode.cn/problems/length-of-last-word/
"""


class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        n = len(s)
        index = n - 1
        # 跳过尾部空格
        while s[index] == ' ':
            index -= 1
        result = 0
        # 从最后一个单词的末尾向前计数
        for i in range(index, -1, -1):
            if s[i] != ' ':
                result += 1
            else:
                break
        return result